napari/components/_layer_slicer.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import annotations
+
+import logging
+from concurrent.futures import Executor, Future, ThreadPoolExecutor
+from threading import RLock
+from typing import Dict, Iterable, Optional, Tuple
+
+from napari.components import Dims
+from napari.layers import Layer
+from napari.utils.events.event import EmitterGroup, Event
+
+logger = logging.getLogger("napari.components._layer_slicer")
+
+
+class _LayerSlicer:
+    """
+    High level class to control the creation of a slice (via a slice request),
+    submit it (synchronously or asynchronously) to a thread pool, and emit the
+    results when complete.
+
+    Events
+    ------
+    ready (value: T)
+        emitted after slicing is done
```

Comment:
We can't add detailed typing information here, but we can add a little to the description so that someone connecting to this knows how to handle the event value.

```suggestion
    ready
        emitted after slicing is done with a dict value that maps from layer to slice response 
```